hummingbot/connector/exchange/chainflip_lp/chainflip_lp_data_formatter.py

```diff
@@ -0,0 +1,282 @@
+                        "base_asset": base_asset,
+                        "quote_asset": quote_asset
+                    })
+                except Exception:
+                    continue
+        return format_list
+    @classmethod
+    def format_all_assets_response(cls,response:Dict):
+        result = response["result"]
+        return result
+    @classmethod
+    def format_error_response(cls, response:Dict):
+        # figure a better way to handle errors
+        return None
+    @classmethod
+    def format_asset_precision(cls,asset:Dict[str,str]):
+        precisions = CONSTANTS.ASSET_PRECISIONS
+        if asset["chain"] in precisions.keys():
+            asset_precision = precisions[asset["chain"]]
+        else:
+            asset_precision = precisions["Ethereum"]
+        if asset["asset"] in CONSTANTS.STABLE_ASSETS:
+            asset_precision = precisions["Stable"]
+        return asset_precision
+        
+    @classmethod
+    def format_orderbook_response(cls,response:Dict, base_asset:Dict[str,str], quote_asset:Dict[str,str]):
+        data = response["result"]
+        bids = list(
+            map(
+                lambda x: {
+                    "amount": cls.format_hex_balance(
+                        x["amount"],
+                        base_asset
+                        ),
+                    "price": cls.format_price(
+                        x["sqrt_price"],
+                        base_asset,
+                        quote_asset,
+                        ),
+                },
+                data["bids"]
+            
+            )
+            )
+        asks = list(
+            map(
+                lambda x: {
+                    "amount": cls.format_hex_balance(
+                        x["amount"],
+                        quote_asset
+                        ),
+                    "price": cls.format_price(
+                        x["sqrt_price"],
+                        base_asset,
+                        quote_asset,
+                    ),
+                },
+                data["asks"]
+            
+            )
+            )
+        format_data = {
+            "bids": bids,
+            "asks": asks
+        }
+        return format_data
+    @classmethod
+    def format_supported_assets(cls, all_pairs:List[Dict[str, str]]):
+        """
+        format the trading pair list of dict to a list of string in
+        the format of {base_asset}-{quote-asset} 
+        """
+        pairs = []
+        for pair in all_pairs:
+            pairs.append(cls.format_assets_to_market_symbol(
+                pair["base_asset"],
+                pair["quote_asset"]
+            ))
+        return pairs
+
+
+    @classmethod
+    def format_trading_pair(cls, pair:str, all_assets:List[Dict[str, str]]):
+        """
+        format a trading pair from {base_asset}-{quote_asset}
+        to the format needed for lp rpc calls
+        e.g ETH-USDT => 
```

Comment:
chainflip supports the same ticker/symbol on multiple chains, eg the `ETH` symbol is available on `Arbitrum` and one `Ethereum`. that is why the rpc methods expect a chain in addition to the symbol. how would this method handle the `Arbitrum.ETH-Ethereum.USDC` trading pair?

Reply:
I thought of this as well but from my knowledge, hummingbot has no implementation for differentiating same asset  symbols but different chain. hummingbot just sends symbols to the connector  and expect a return data from those symbols. it is the way normal centralized exchanges work. chainflip Lp is a unique implementation in the since that it is a decentralized exchange that works in a centralized exchanges way so there going to be set backs when trying to use full centralized logic on it. I guess this is one of it. My suggestion is that all `ETH` and Stable coins pairs should be traded on the normal `Ethereum` chain when using hummingbot .

Reply:
i see, that is unfortunate. but maybe it makes sense to allow configuring the default chain for a symbol? eg configure that the integration should always use `Arbitrum` for the `USDC` symbol and `Ethereum` for the `ETH` symbol

Reply:
@niklasnatter not sure if it works (I was reading https://github.com/chainflip-io/chainflip-backend/pull/4614/files but not 100% sure about it, I dont read rust that well), but would passing in "`arbUSDC`" and "`arbETH`" serialize correctly? Or are only the legacy ones supported?

Reply:
no - the rpc methods only allow the `{chain, asset}` encoding for newer assets, but i think that is not the problem here.  if i understand right, the problem is that hummingbot does not expect multiple assets for the same symbol (`ETH`) because centralised exchanges do not have this